### PR TITLE
add missing extensions required by Sphinx & pytest to Python 3.7.{2,4}

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/p/Python/Python-3.7.2-GCCcore-8.2.0.eb [hunk 1]
```diff
@@ -42,6 +42,7 @@ installopts = " && ln -s %(installdir)s/bin/pip3 %(installdir)s/bin/pip"
 exts_default_options = {
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/p/Python/Python-3.7.2-GCCcore-8.2.0.eb]: missing `'source_urls': [PYPI_SOURCE]`, if they are not specified for each extension

Comment (Flamefire) [reply easybuild/easyconfigs/p/Python/Python-3.7.2-GCCcore-8.2.0.eb]: Thanks. I added it here and removed it from the dict below to unify with the other EC

**Comment by boegel** [comment]
@Flamefire Looks OK for Python 3.7.4 (`pip check` passes), but not for Python 3.7.2:

```
== 2019-11-19 22:35:52,794 build_log.py:164 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): Sanity check failed: extensions sanity check failed for 5 extensions: sphinxcontrib-applehelp, sphinxcontrib-devhelp, sphinxcontrib-htmlhelp, sphinxcontrib-qthelp, sphinxcontrib-serializinghtml
failing sanity check for 'sphinxcontrib-applehelp' extension: command "python -c "import sphinxcontrib.applehelp"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinxcontrib/applehelp/__init__.py", line 24, in <module>
    from sphinx.util import SkipProgressMessage, progress_message
ImportError: cannot import name 'SkipProgressMessage' from 'sphinx.util' (/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinx/util/__init__.py),
failing sanity check for 'sphinxcontrib-devhelp' extension: command "python -c "import sphinxcontrib.devhelp"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinxcontrib/devhelp/__init__.py", line 25, in <module>
    from sphinx.util.nodes import NodeMatcher
ImportError: cannot import name 'NodeMatcher' from 'sphinx.util.nodes' (/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinx/util/nodes.py),
failing sanity check for 'sphinxcontrib-htmlhelp' extension: command "python -c "import sphinxcontrib.htmlhelp"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinxcontrib/htmlhelp/__init__.py", line 28, in <module>
    from sphinx.util import progress_message
ImportError: cannot import name 'progress_message' from 'sphinx.util' (/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinx/util/__init__.py),
failing sanity check for 'sphinxcontrib-qthelp' extension: command "python -c "import sphinxcontrib.qthelp"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinxcontrib/qthelp/__init__.py", line 26, in <module>
    from sphinx.util.nodes import NodeMatcher
ImportError: cannot import name 'NodeMatcher' from 'sphinx.util.nodes' (/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinx/util/nodes.py),
failing sanity check for 'sphinxcontrib-serializinghtml' extension: command "python -c "import sphinxcontrib.serializinghtml"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinxcontrib/serializinghtml/__init__.py", line 16, in <module>
    from sphinx.deprecation import RemovedInSphinx40Warning, deprecated_alias
ImportError: cannot import name 'deprecated_alias' from 'sphinx.deprecation' (/software/Python/3.7.2-GCCcore-8.2.0/lib/python3.7/site-packages/sphinx/deprecation.py),  (at easybuild/framework/easyblock.py:2592 in _sanity_check_step)
```

**Comment by Flamefire** [comment]
Ok so am I seeing this correctly?:
- sphinx depends on sphinxcontrib-applehelp
- sphinxcontrib-applehelp uses sphinx

On the actual error: Seems 3.7.2 uses Sphinx 1.x which only requires `sphinxcontrib-websupport`

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/722f04dfef85eb3a1d09dd461bbf5f0c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/b679d0d2522877256181ccdf3ed87783 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3124.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/5dab9b97ecf0c06c85e44339dec9d42e for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
taurusi6296.taurus.hrsk.tu-dresden.de - Linux RHEL 7.4, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e856aef04470397ade57f7307158910c for a full test report.

**Pull request merged**